os/binfmt/elf.c

```diff
@@ -74,11 +74,11 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-/* CONFIG_DEBUG, CONFIG_DEBUG_INFO, and CONFIG_DEBUG_BINFMT have to be
+/* CONFIG_DEBUG, CONFIG_DEBUG_VERBOSE, and CONFIG_DEBUG_BINFMT have to be
  * defined or CONFIG_ELF_DUMPBUFFER does nothing.
  */
 
-#if !defined(CONFIG_DEBUG_INFO) || !defined(CONFIG_DEBUG_BINFMT)
+#if !defined(CONFIG_DEBUG_BINFMT_INFO)
```

Comment:
Is this necessary to be two steps?
```c
#if !defined(CONFIG_DEBUG_BINFMT_INFO)
#undef CONFIG_ELF_DUMPBUFFER
#endif

#ifdef CONFIG_ELF_DUMPBUFFER
#define elf_dumpbuffer(m, b, n) binfodumpbuffer(m, b, n)
#else
#define elf_dumpbuffer(m, b, n)
#endif
```

How about checking it directly?
```c
#ifdef CONFIG_DEBUG_BINFMT_INFO
#define elf_dumpbuffer(m, b, n) binfodumpbuffer(m, b, n)
#else
#define elf_dumpbuffer(m, b, n)
#endif
```

Reply:
if CONFIG_DEBUG_BINFMT_INFO enabled and CONFIG_ELF_DUMPBUFFER is not, then it should not set elf_dumpbuffer, so I think the current approach is fine

Reply:
or we can replace with :
#ifdef CONFIG_DEBUG_BINFMT_INFO
#ifdef CONFIG_ELF_DUMPBUFFER
#define elf_dumpbuffer(m, b, n) binfodumpbuffer(m, b, n)
#else
#define elf_dumpbuffer(m, b, n)
#endif
#endif

Reply:
os/binfmt/libelf/Kconfig
```kconfig
config ELF_DUMPBUFFER
	bool "Dump ELF buffers"
	default n
	depends on DEBUG_BINFMT_INFO  <---- here
	---help---
		Dump various ELF buffers for debug purposes
```

in here, os/binfmt/elf.c
```c
remove 1st conditional  <--- here

#ifdef CONFIG_ELF_DUMPBUFFER
#define elf_dumpbuffer(m, b, n) binfodumpbuffer(m, b, n)
#else
#define elf_dumpbuffer(m, b, n)
#endif
```

This is enough and is the same as current pr

Reply:
done